src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -559,28 +591,104 @@ Namespace Microsoft.CodeAnalysis.Semantics
+            Dim isImplicit As Boolean = boundConversion.WasCompilerGenerated OrElse Not boundConversion.ExplicitCastInCode
+
+            Dim conversionInformation = CreateConversionOperation(boundConversion.Operand, boundConversion.ConversionKind, boundConversion.Syntax)
+            Dim methodSymbol As MethodSymbol = conversionInformation.methodSymbol
+            Dim operand As Lazy(Of IOperation) = conversionInformation.operation
+            Dim isAddressOfDelegateCreation As Boolean = conversionInformation.isAddressOfDelegateCreation
+
+            If isAddressOfDelegateCreation OrElse
+               ((boundConversion.Operand.Kind = BoundKind.Lambda OrElse
+                 boundConversion.Operand.Kind = BoundKind.QueryLambda OrElse
+                 boundConversion.Operand.Kind = BoundKind.UnboundLambda) AndAlso
+                boundConversion.Type.IsDelegateType()) Then
+                ' If this is a conversion from a lambda to a delegate type, or this is an AddressOf delegate creation
+                ' as determined above, we return a DelegateCreationExpression, instead of returning a conversion expression
+                Return New LazyDelegateCreationExpression(operand, _semanticModel, syntax, type, constantValue, isImplicit)
             End If
 
-            Dim conversion = New Conversion(New KeyValuePair(Of ConversionKind, MethodSymbol)(boundConversion.ConversionKind, methodSymbol))
+            Dim conversion = New Conversion(New KeyValuePair(Of VisualBasic.ConversionKind, MethodSymbol)(boundConversion.ConversionKind, methodSymbol))
             Dim isExplicit As Boolean = boundConversion.ExplicitCastInCode
             Dim isTryCast As Boolean = False
             Dim isChecked As Boolean = False
-            Dim type As ITypeSymbol = boundConversion.Type
-            Dim constantValue As [Optional](Of Object) = ConvertToOptional(boundConversion.ConstantValueOpt)
-            Dim isImplicit As Boolean = boundConversion.WasCompilerGenerated
             Return New LazyVisualBasicConversionExpression(operand, conversion, isExplicit, isTryCast, isChecked, _semanticModel, syntax, type, constantValue, isImplicit)
         End Function
 
+        Private Function IsDelegateCreationSyntax(conversionKind As ConversionKind, conversionSyntax As SyntaxNode, operandSyntax As SyntaxNode) As Boolean
+            ' An identity conversion is introduced by the compiler on top of New DelegateType(AddressOf Method). This node
+            ' only exists for the convenience of SemanticModel, so we want to classify it as a Delegate Creation
+            Dim validConversionKind = conversionKind = ConversionKind.Identity OrElse
+                                      (conversionKind And (Not ConversionKind.DelegateRelaxationLevelMask)) = 0
+
+            ' Any of the explicit cast types, as well as New DelegateType(AddressOf Method)
+            Dim validConversionSyntax = conversionSyntax.Kind() = SyntaxKind.CTypeExpression OrElse
```

Comment:
>validConversionSyntax [](start = 16, length = 21)

I guess we are missing the case of an implicit conversion, i.e. when the conversion node and the operand are both associated with the same AddressOfExpressionSyntax

Reply:
Added.

---
In reply to: [142996089](https://github.com/dotnet/roslyn/pull/22056#discussion_r142996089) [](ancestors = 142996089)